test/src/test/java/hudson/model/ProjectTest.java

```diff
@@ -175,9 +175,9 @@ public void testGetAssignedLabel() throws Exception{
     public void testGetAssignedLabelString() throws Exception{
         FreeStyleProject p = j.createFreeStyleProject("project");
         Slave slave = j.createOnlineSlave();
-        assertNull("Project should not have any label.", p.getAssignedLabelString());
+        assertEquals("Project should have jenkins master label.", j.jenkins.getSelfLabel().name, p.getAssignedLabelString());
```

Comment:
jtnord, I'm not familiar with the behavior described here - where a project with no node assigned has different tied-label behavior depending on whether Jenkins has slaves or not. 

I just looked at uses of the canRoam field - it is both private and volatile.    the 'setAssignedLabel' method does not take the slave count into consideration when setting canRoam to true or false.    It just checks whether the provided label argument is null or non-null.